x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityRestIT.java

```diff
@@ -237,31 +307,89 @@ public void testRemoteAccessForCrossClusterSearch() throws Exception {
-                )
+                sendRequestWith.equals("user")
+                    ? containsString(
+                        "action [indices:data/read/search] towards remote cluster [my_remote_cluster]"
+                            + " is unauthorized for user [local_search_user] with effective roles [local_search]"
+                            + " because no remote indices privileges apply for the target cluster"
+                    )
+                    : containsString(
+                        "action [indices:data/read/search] towards remote cluster [my_remote_cluster] "
+                            + "is unauthorized for API key id ["
+                            + localApiKeyId
+                            + "] of user [local_search_user] because no remote indices privileges apply for the target cluster"
+                    )
             );
 
             // Check that authentication fails if we use a non-existent API key
-            updateClusterSettings(Settings.builder().put("cluster.remote.my_remote_cluster.authorization", randomEncodedApiKey()).build());
+            updateClusterSettings(
+                Settings.builder()
+                    .put(
+                        "cluster.remote.my_remote_cluster.authorization",
+                        randomValueOtherThan(remoteAccessApiKeyId, () -> randomEncodedApiKey())
+                    )
+                    .build()
+            );
             final ResponseException exception4 = expectThrows(
                 ResponseException.class,
-                () -> performRequestWithRemoteAccessUser(new Request("GET", "/my_remote_cluster:index1/_search"))
+                sendRequestWith.equals("user")
+                    ? () -> performRequestWithRemoteAccessUser(new Request("GET", "/my_remote_cluster:index1/_search"))
+                    : () -> performRequestWithApiKey(new Request("GET", "/my_remote_cluster:index1/_search"), localApiKeyEncoded)
             );
-            assertThat(exception4.getResponse().getStatusLine().getStatusCode(), equalTo(401));
-            assertThat(exception4.getMessage(), containsString("unable to authenticate user"));
-            assertThat(exception4.getMessage(), containsString("unable to find apikey"));
+            if (sendRequestWith.equals("user")) {
+                assertThat(exception4.getResponse().getStatusLine().getStatusCode(), equalTo(401));
+                assertThat(exception4.getMessage(), containsString("unable to authenticate user"));
+                assertThat(exception4.getMessage(), containsString("unable to find apikey"));
+            } else {
+                assertThat(exception4.getResponse().getStatusLine().getStatusCode(), equalTo(403));
+                assertThat(exception4.getMessage(), containsString("unauthorized for API key id [" + localApiKeyId + "]"));
+                assertThat(exception4.getMessage(), containsString("no remote indices privileges apply for the target cluster"));
+            }
```

Comment:
Optional but would also cover the case where a QC-side user has access to an index but their QC-side API key does not.